### PR TITLE
[Improve] Support using multiple hadoop account

### PR DESCRIPTION
**Pull request by ruanwenjun**
### Purpose of this pull request

Refactor the usage of `UserGroupInformation` don't use `setLoginUser` to login in. This method will set a static user in jvm, if and cause we cannot support use multiple hadoop account in one seatunnel server.

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by Hisoka-X — COMMENTED** [review]
Please add test case

**Comment by ruanwenjun** [comment]
> Please add test case

Add e2e for kerberos, this pr is compatible with the current version.

**Review by hailin0 — DISMISSED** [review]

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/WriteStrategy.java [hunk 1]
```diff
@@ -99,12 +99,5 @@ public interface WriteStrategy extends Transaction, Serializable {
...
      */
-    FileSystemUtils getFileSystemUtils();
-
-    /**
-     * set file system utils
-     *
-     * @param fileSystemUtils fileSystemUtils
-     */
-    void setFileSystemUtils(FileSystemUtils fileSystemUtils);
+    HadoopFileSystemProxy getFileSystemUtils();
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/WriteStrategy.java]:
```java
     * @return file system utils
     */
    HadoopFileSystemProxy getFileSystemUtils();
```
=>
```java
     * @return hadoop file system proxy
     */
    HadoopFileSystemProxy getHadoopFileSystemProxy();
```

Comment (ruanwenjun) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/WriteStrategy.java]: Done

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+        if (enableRemoteUser()) {
+            return HadoopLoginFactory.loginWithRemoteUser(
+                    configuration, hadoopConf.getRemoteUser(), loginFunction);
+        }
+        return loginFunction.run(configuration, UserGroupInformation.getCurrentUser());
+    }
+
+    @Override
+    public void close() throws IOException {
+        synchronized (UserGroupInformation.class) {
```
Comment (Carl-Zhou-CN) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java]: I think there is a lock based on UserGroupInformation within the logoutUserFromKeytab method. Do we need to expand the scope of the lock?

Comment (ruanwenjun) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java]: Good catch, this lock is not needed, I removed this lock.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+                String.format("fs.%s.impl", hadoopConf.getSchema()), hadoopConf.getFsHdfsImpl());
+        hadoopConf.setExtraOptionsForConfiguration(configuration);
+        return configuration;
+    }
+
+    private boolean enableKerberos() {
+        boolean kerberosPrincipalEmpty = StringUtils.isEmpty(hadoopConf.getKerberosPrincipal());
+        boolean kerberosKeytabPathEmpty = StringUtils.isEmpty(hadoopConf.getKerberosKeytabPath());
+        boolean krb5FilePathEmpty = StringUtils.isEmpty(hadoopConf.getKrb5Path());
+        if (kerberosKeytabPathEmpty && kerberosPrincipalEmpty && krb5FilePathEmpty) {
```
Comment (Carl-Zhou-CN) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java:262]: krb5FilePathEmpty is it necessarily FALSE,i think he has a default value?

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/hive/HiveJdbcConnectionProvider.java [hunk 4]
```diff
@@ -76,4 +62,43 @@ public Connection getOrEstablishConnection() throws SQLException, ClassNotFoundE
...
+        }
+
+        public Connection produce() throws SQLException {
+            final Properties info = new Properties();
+            jdbcConnectionConfig
+                    .getUsername()
+                    .ifPresent(username -> info.setProperty("user", username));
+            jdbcConnectionConfig
+                    .getPassword()
+                    .ifPresent(username -> info.setProperty("password", username));
```
Comment (Carl-Zhou-CN) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/hive/HiveJdbcConnectionProvider.java:100]:
```suggestion
                   .ifPresent(password-> info.setProperty("password", password));
```

Comment (Carl-Zhou-CN) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/hive/HiveJdbcConnectionProvider.java:100]: I wonder if 'user','password' would be better with values from JdbcOptions?

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java (hunk 3 above)
Comment (ruanwenjun) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java:262]: Yes, we have default value in options.

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/hive/HiveJdbcConnectionProvider.java (hunk 4 above)
Comment (ruanwenjun) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/hive/HiveJdbcConnectionProvider.java:100]: Good cache, in fact the value of `user` and `password` is from `JdbcOptions `.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java (hunk 3 above)
Comment (Carl-Zhou-CN) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java:262]: If krb5FilePathEmpty must be false, an exception seems to be thrown when kerberos authentication is not enabled

Comment (ruanwenjun) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java:262]: We will set the value to `/etc/krb5.conf` if user doesn't set this value in conf, but if you want to use `HadoopFileSystemProxy` in code, there is no default value.

Comment (Carl-Zhou-CN) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java:262]: ok

**Review by Carl-Zhou-CN — APPROVED** [review]
LGTM

**Review by Hisoka-X — COMMENTED** [review]
Overall LGTM. cc @TyrantLucifer

**Review by hailin0 — APPROVED** [review]

**Pull request merged**